std/codata/pi.pol

```diff
@@ -0,0 +1,6 @@
+use "./fun.pol"
+
+-- | The dependent function type.
+codata Π(a: Type, p: a -> Type) {
```

Comment:
Do we want to use type-theory notation or more applied terms? E.g. we could also call `Π` `Forall` with syntax sugar `∀`.

Reply:
I will rename it to `Pi` without unicode syntax, but Forall might also refer to the type specialized to `a := Type`, i.e. `codata Forall(p : Type -> Type) { ... }`.